log-manager/src/main/java/io/airlift/log/LogFileName.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.log;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ComparisonChain;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import static com.google.common.base.CharMatcher.anyOf;
+import static java.lang.Integer.parseInt;
+import static java.util.Objects.requireNonNull;
+import static java.util.UUID.randomUUID;
+
+class LogFileName
+        implements Comparable<LogFileName>
+{
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("-yyyyMMdd.HHmmss");
+    private static final int MAX_GENERATED_INDEX = 1000;
+
+    private final String fileName;
+    private final LocalDateTime dateTime;
+    private final OptionalInt index;
+    private final OptionalInt legacyIndex;
+    private final Optional<String> slug;
```

Comment:
what the slug is for? 
if it's for differentiating different log streams, then i think it should come at the beginning, not close to the end?

Reply:
The slug is allowed sometime present when there are too man name collisions.  This should never actually happen, but we need to fail safe.  It goes at the end because it is last in the sort order.